app/ui/browser/reducers/main-reducers.js

```diff
@@ -37,15 +37,26 @@ function getPageIndexById(state, id) {
 export default function basic(state = initialState, action) {
   switch (action.type) {
     case types.CREATE_TAB:
```

Comment:
I guess it would probably be too slow to go over IPC for the contextual menu New Tab actions? I don't know much about the cost of IPC.